### DISCUSSION TITLE
How do I use SQLAlchemy to update a list of rows with data that is provided via JSON?

### DISCUSSION THREAD

**Discussion by moonman239** [sqlalchemy/sqlalchemy#8747] [Usage Questions]
I have a Flask app, with an endpoint that is for updating records. The client shall pass data to the endpoint, and then the endpoint shall update the relevant rows in the database.

For example, the client might send ```[{"id":1,"calories":1800},{"id":2,"calories":1810}]``` and then my Flask app shall deserialize that JSON and shall use SQLAlchemy to tell the RDBMS to update records with the indicated ids to have the indicated calorie counts.

I have an ORM object that looks like this:

```
class Person(Base):
    id = Column(Integer,primary_key=True)
    calories = Column(Integer)
```

I know how to deserialize JSON into any object in Marshmallow, but I'm wondering what the SQLAlchemy-esque way of accomplishing the titular goal is.

**Comment by zzzeek** [comment]
the most direct way is just to feed that JSON directly into a [bulk update](https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.Session.bulk_update_mappings):

    session.bulk_update_mappings(Person, [{"id":1,"calories":1800},{"id":2,"calories":1810}])

now on those docs, which are 1.4, you will see a lot of warnings.   However, when you upgrade to 2.0, you can take the above statement and use a much nicer format of it which is upcoming at [bulk update by primary key](https://docs.sqlalchemy.org/en/20/orm/queryguide/dml.html#orm-queryguide-bulk-update)
